includes/class-wc-session-handler.php

```diff
@@ -93,8 +93,8 @@ public function init_session_cookie() {
 			$this->_data               = $this->get_session_data();
 
 			// If the user logs in, update session.
-			if ( is_user_logged_in() && get_current_user_id() !== $this->_customer_id ) {
-				$this->_customer_id = get_current_user_id();
+			if ( is_user_logged_in() && strval( get_current_user_id() ) !== $this->_customer_id ) {
```

Comment:
Instead of converting to string, it should be converted to integer:

```php
if ( is_user_logged_in() && get_current_user_id() !== intval( $this->_customer_id ) ) {
```

Reply:
That's what I originally did,  but then I noticed line 181:

```php
$customer_id = md5( $hasher->get_random_bytes( 32 ) );
```

If a user has just logged in then `$this->_customer_id` is going to be an md5 hash and `intval` will interpret that as an integer.  What if the logged in user has as id of 57 and the md5 hash just happened to be say `57e3119fb084fe8bbf5fe3ab7cc89b3b`?

Maybe the comparison should be with `strval` and the assignment below as `intval`?

Or maybe it needs an even bigger re-factoring...

Reply:
After reading the whole code (methods like generate_customer_id() that excepts strings) I think strings should works just fine.